--- AR6kSDK.build_sw.83/host/os/linux/ar6000_drv.c.orig	2024-05-08 07:09:19.554141653 -0400
+++ AR6kSDK.build_sw.83/host/os/linux/ar6000_drv.c	2024-05-08 10:03:55.526185508 -0400
@@ -2067,7 +2067,7 @@
 ar6000_stop_endpoint(struct net_device *dev)
 {
     AR_SOFTC_T *ar = (AR_SOFTC_T*)netdev_priv(dev);
-    printk("%s -- enter\n", __func__);
+    AR_DEBUG_PRINTF("%s -- enter\n", __func__);
     /* Stop the transmit queues */
     netif_stop_queue(dev);
 
@@ -4146,7 +4146,7 @@
         A_PRINTF(" from %2.2x:%2.2x:%2.2x:%2.2x:%2.2x:%2.2x ",
                  bssid[0], bssid[1], bssid[2], bssid[3], bssid[4], bssid[5]);
     }
-
+    A_PRINTF("\n");
     AR_DEBUG_PRINTF("\nDisconnect Reason is %d", reason);
     AR_DEBUG_PRINTF("\nProtocol Reason/Status Code is %d", protocolReasonStatus);
     AR_DEBUG_PRINTF("\nAssocResp Frame = %s",
