diff --git a/slimproto.c b/slimproto.c
index 7b82869..3685acf 100644
--- a/slimproto.c
+++ b/slimproto.c
@@ -114,7 +114,7 @@ void send_packet(u8_t *packet, size_t len) {
 }
 
 static void sendHELO(bool reconnect, const char *fixed_cap, const char *var_cap, u8_t mac[6]) {
-	#define BASE_CAP "Model=squeezelite,AccuratePlayPoints=1,HasDigitalOut=1,HasPolarityInversion=1,Balance=1,Firmware=" VERSION
+	#define BASE_CAP "Model=baby,AccuratePlayPoints=1,HasPolarityInversion=1,Balance=1,Firmware=" VERSION
 	#define SSL_CAP "CanHTTPS=1"
 	const char *base_cap;
 	struct HELO_packet pkt;
