--- fdk-aac-2.0.3/Makefile.am.orig	2023-12-21 12:56:58.000000000 -0500
+++ fdk-aac-2.0.3/Makefile.am	2024-11-03 15:05:32.108443658 -0500
@@ -244,12 +244,12 @@
     libSYS/src/syslib_channelMapDescr.cpp
 
 libfdk_aac_la_SOURCES = \
-    $(AACDEC_SRC) $(AACENC_SRC) \
+    $(AACDEC_SRC) \
     $(ARITHCODING_SRC) \
     $(DRCDEC_SRC) \
-    $(MPEGTPDEC_SRC) $(MPEGTPENC_SRC) \
-    $(SACDEC_SRC) $(SACENC_SRC) \
-    $(SBRDEC_SRC) $(SBRENC_SRC) \
+    $(MPEGTPDEC_SRC) \
+    $(SACDEC_SRC) \
+    $(SBRDEC_SRC) \
     $(PCMUTILS_SRC) $(FDK_SRC) $(SYS_SRC)
 
 EXTRA_DIST = \
