--- faad2-2.10.1/libfaad/Makefile.am.orig	2022-10-20 15:59:26.000000000 -0400
+++ faad2-2.10.1/libfaad/Makefile.am	2024-11-10 15:36:22.924771362 -0500
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = libfaad.la libfaad_drm.la
+lib_LTLIBRARIES = libfaad.la
 
 AM_CPPFLAGS	= -iquote $(top_srcdir)/include
 include_HEADERS = $(top_srcdir)/include/faad.h \
@@ -9,7 +9,7 @@
 libfaad_la_CFLAGS = -fvisibility=hidden
 
 libfaad_la_SOURCES = bits.c cfft.c decoder.c drc.c \
-		     drm_dec.c error.c filtbank.c \
+		     error.c filtbank.c \
 		     ic_predict.c is.c lt_predict.c mdct.c mp4.c ms.c output.c pns.c \
 		     ps_dec.c ps_syntax.c \
 		     pulse.c specrec.c syntax.c tns.c hcr.c huffman.c \
@@ -17,7 +17,7 @@
 		     sbr_dct.c sbr_e_nf.c sbr_fbt.c sbr_hfadj.c sbr_hfgen.c \
 		     sbr_huff.c sbr_qmf.c sbr_syntax.c sbr_tf_grid.c sbr_dec.c \
 		     analysis.h bits.h cfft.h cfft_tab.h common.h \
-		     drc.h drm_dec.h error.h fixed.h filtbank.h \
+		     drc.h error.h fixed.h filtbank.h \
 		     huffman.h ic_predict.h iq_table.h is.h kbd_win.h lt_predict.h \
 		     mdct.h mdct_tab.h mp4.h ms.h output.h pns.h ps_dec.h ps_tables.h \
 		     pulse.h rvlc.h \
@@ -30,10 +30,10 @@
 		     codebook/hcb_5.h codebook/hcb_6.h codebook/hcb_7.h codebook/hcb_8.h \
 		     codebook/hcb_9.h codebook/hcb_10.h codebook/hcb_11.h codebook/hcb_sf.h
 
-libfaad_drm_la_LDFLAGS = ${libfaad_la_LDFLAGS}
-libfaad_drm_la_LIBADD = ${libfaad_la_LIBADD}
-libfaad_drm_la_CFLAGS = ${libfaad_la_CFLAGS} -DDRM_SUPPORT
-libfaad_drm_la_SOURCES = ${libfaad_la_SOURCES}
+libfaad_drm_la_LDFLAGS =
+libfaad_drm_la_LIBADD =
+libfaad_drm_la_CFLAGS =
+libfaad_drm_la_SOURCES =
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = faad2.pc
